### ISSUE TITLE
Creating a new project or a Gem via Command Line or Terminal causes errors to be printed

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#10515]
**Describe the bug**
If a new project or a new Gem is created via the Command Line or the Terminal, the following error messages are printed to the log:
```
[ERROR] o3de.manifest: A cycle has been detected when visiting external subdirectories at gem path "C:/o3de/Gems/Atom/RHI/DX12". The visited paths are: {'C:/o3de/Gems/Atom/Bootstrap', 'C:/o3de/Gems/Atom/RHI/DX12', 'C:/o3de/Gems/Atom/RHI/Metal', 'C:/o3de/Gems/Atom/Tools/AtomToolsFramework', 'C:/o3de/Gems/Atom/Tools/MaterialEditor', 'C:/o3de/Gems/Atom', 'C:/o3de/Gems/Atom/RHI', 'C:/o3de/Gems/Atom/Component/DebugCamera', 'C:/o3de/Gems/Atom/RHI/Null', 'C:/o3de/Gems/Atom/Tools/ShaderManagementConsole', 'C:/o3de/Gems/Atom/Asset/Shader', 'C:/o3de/Gems/Atom/Asset/ImageProcessingAtom', 'C:/o3de/Gems/Atom/RPI', 'C:/o3de/Gems/Atom/RHI/Vulkan', 'C:/o3de/Gems/Atom/Feature/Common', 'C:/o3de/Gems/Atom/Tools/MaterialCanvas'}
[ERROR] o3de.manifest: A cycle has been detected when visiting external subdirectories at gem path "C:/o3de/Gems/Atom/RHI/Metal". The visited paths are: {'C:/o3de/Gems/Atom/Bootstrap', 'C:/o3de/Gems/Atom/RHI/DX12', 'C:/o3de/Gems/Atom/RHI/Metal', 'C:/o3de/Gems/Atom/Tools/AtomToolsFramework', 'C:/o3de/Gems/Atom/Tools/MaterialEditor', 'C:/o3de/Gems/Atom', 'C:/o3de/Gems/Atom/RHI', 'C:/o3de/Gems/Atom/Component/DebugCamera', 'C:/o3de/Gems/Atom/RHI/Null', 'C:/o3de/Gems/Atom/Tools/ShaderManagementConsole', 'C:/o3de/Gems/Atom/Asset/Shader', 'C:/o3de/Gems/Atom/Asset/ImageProcessingAtom', 'C:/o3de/Gems/Atom/RPI', 'C:/o3de/Gems/Atom/RHI/Vulkan', 'C:/o3de/Gems/Atom/Feature/Common', 'C:/o3de/Gems/Atom/Tools/MaterialCanvas'}
[ERROR] o3de.manifest: A cycle has been detected when visiting external subdirectories at gem path "C:/o3de/Gems/Atom/RHI/Null". The visited paths are: {'C:/o3de/Gems/Atom/Bootstrap', 'C:/o3de/Gems/Atom/RHI/DX12', 'C:/o3de/Gems/Atom/RHI/Metal', 'C:/o3de/Gems/Atom/Tools/AtomToolsFramework', 'C:/o3de/Gems/Atom/Tools/MaterialEditor', 'C:/o3de/Gems/Atom', 'C:/o3de/Gems/Atom/RHI', 'C:/o3de/Gems/Atom/Component/DebugCamera', 'C:/o3de/Gems/Atom/RHI/Null', 'C:/o3de/Gems/Atom/Tools/ShaderManagementConsole', 'C:/o3de/Gems/Atom/Asset/Shader', 'C:/o3de/Gems/Atom/Asset/ImageProcessingAtom', 'C:/o3de/Gems/Atom/RPI', 'C:/o3de/Gems/Atom/RHI/Vulkan', 'C:/o3de/Gems/Atom/Feature/Common', 'C:/o3de/Gems/Atom/Tools/MaterialCanvas'}
[ERROR] o3de.manifest: A cycle has been detected when visiting external subdirectories at gem path "C:/o3de/Gems/Atom/RHI/Vulkan". The visited paths are: {'C:/o3de/Gems/Atom/Bootstrap', 'C:/o3de/Gems/Atom/RHI/DX12', 'C:/o3de/Gems/Atom/RHI/Metal', 'C:/o3de/Gems/Atom/Tools/AtomToolsFramework', 'C:/o3de/Gems/Atom/Tools/MaterialEditor', 'C:/o3de/Gems/Atom', 'C:/o3de/Gems/Atom/RHI', 'C:/o3de/Gems/Atom/Component/DebugCamera', 'C:/o3de/Gems/Atom/RHI/Null', 'C:/o3de/Gems/Atom/Tools/ShaderManagementConsole', 'C:/o3de/Gems/Atom/Asset/Shader', 'C:/o3de/Gems/Atom/Asset/ImageProcessingAtom', 'C:/o3de/Gems/Atom/RPI', 'C:/o3de/Gems/Atom/RHI/Vulkan', 'C:/o3de/Gems/Atom/Feature/Common', 'C:/o3de/Gems/Atom/Tools/MaterialCanvas'}
```
It should be noted that the created project, and Gem do not have any apparent issues.

**Prerequisites**
Cloned, and registered O3DE engine

**Steps to reproduce**
1. Launch the Command Line/Terminal and navigate to the `o3de` directory.
2. Execute either the `scripts\o3de.bat create-project -pp <new project path>` (for example: `scripts\o3de.bat create-project -pp C:\ExampleProject`) or `scripts\o3de.bat create-gem -gp <new Gem path>` (for example: `scripts\o3de.bat create-gem -gp C:\ExampleGem`).

**Expected behavior**
Commands execute without any errors.

**Actual behavior**
Errors are printed to the Command Line/Terminal.

**Screenshot and log**
![CMDCreationScreenshot](https://user-images.githubusercontent.com/86952082/177157721-c872e006-4ef3-4982-9b86-a713fbc74478.PNG)
[CMDCreationLog.txt](https://github.com/o3de/o3de/files/9039504/CMDCreationLog.txt)

**Found in Branch**
Development (db9d67d)

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by AMZN-alexpete** [comment]
Looks like a bug in the cycle detection I added recently, or we should silence the error because it currently doesn't prevent project or gem creation from working

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (ff77e99) build.
